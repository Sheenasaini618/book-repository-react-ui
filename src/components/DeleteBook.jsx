import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import DeleteIcon from '@mui/icons-material/Delete';
import IconButton from '@mui/material/IconButton';
import { deleteBook } from '../util/api/bookApi';

export default function DeleteBook(props) {
  const [open, setOpen] = React.useState(false);
  const bookId = props.details

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleDelete = () => {
    deleteBook(bookId).then(res => {
      console.log(res)
      handleClose();
      window.location.reload(false);
    })
  };

  return (
    <div>
      <IconButton className = "icon" onClick={handleClickOpen}>
          <DeleteIcon />
      </IconButton>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="responsive-dialog-title"
      >
        <DialogTitle id="responsive-dialog-title">
          {"Are you sure to delete the book?"}
        </DialogTitle>
        <DialogActions>
          <Button className ='closeButton' onClick={handleClose}>
            No
          </Button>
          <Button className ='deleteButton' onClick={handleDelete} autoFocus>
            Yes
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
