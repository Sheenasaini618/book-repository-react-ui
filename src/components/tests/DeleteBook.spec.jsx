import '@testing-library/jest-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import DeleteBook from '../DeleteBook';


configure({ adapter: new Adapter() });

global.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
        ok: true,
        status: 200,
    }),
);
describe('<DeleteBook />', () => {

    let props, component;

    if (typeof window === 'undefined') {
        global.window = {}
    }

    beforeEach(() => {
        props = {
            details: 1
        };

        Object.defineProperty(window, 'location', {
            configurable: true,
            value: { reload: jest.fn() },
        });

        component = shallow(<DeleteBook {...props} />);

    })

    it('should check for icon element', () => {

        const event = {};
        expect(component.find('.icon').length).toBe(1);
        component
            .find('.icon')
            .at(0)
            .simulate('click', event);
    })

    it('should match snapshot', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });

    it('should validate elements in delete book component', () => {
        const event = {};
        expect(component.find('.icon').length).toBe(1);
        component.find('.closeButton').at(0).simulate('click', event);
        component.find('.deleteButton').at(0).simulate('click', event);
    })
})
