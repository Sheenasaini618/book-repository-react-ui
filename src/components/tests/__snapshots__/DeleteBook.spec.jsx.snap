// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteBook /> should match snapshot 1`] = `
"<div>
  <ForwardRef(IconButton) className=\\"icon\\" onClick={[Function: handleClickOpen]}>
    <Memo(ForwardRef(DeleteIcon)) />
  </ForwardRef(IconButton)>
  <ForwardRef(Dialog) open={false} onClose={[Function: handleClose]} aria-labelledby=\\"responsive-dialog-title\\">
    <ForwardRef(DialogTitle) id=\\"responsive-dialog-title\\">
      Are you sure to delete the book?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) className=\\"closeButton\\" onClick={[Function: handleClose]}>
        No
      </ForwardRef(Button)>
      <ForwardRef(Button) className=\\"deleteButton\\" onClick={[Function: handleDelete]} autoFocus={true}>
        Yes
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</div>"
`;
